main.nf

```diff
@@ -1131,112 +1131,119 @@ process busco_download_db {
+    set val(assembler), val(sample), file("short_summary.specific.${db}.${bin}.txt") into (ch_busco_multiqc, ch_busco_to_summary, ch_busco_plot)
+    file("${bin}_busco.log")
+    file("${bin}_buscos.faa") optional true
+    file("${bin}_buscos.fna") optional true
 
     script:
-    if( workflow.profile.toString().indexOf("conda") == -1) {
-        """
+    if( workflow.profile.toString().indexOf("conda") == -1)
+        cp_augustus_config = "Y"
+    else
+        cp_augustus_config = "N"
+
+    """
+    # get path to custom config file for busco (already configured during installation)
+    busco_path=\$(which busco)
+    config_file="\${busco_path%bin/busco}config/config.ini"
+
+    if [ ${cp_augustus_config} = "Y" ] ; then
         cp -r /opt/conda/pkgs/augustus*/config augustus_config/
         export AUGUSTUS_CONFIG_PATH=augustus_config
+    fi
 
-        run_BUSCO.py \
-            --in ${assembly} \
-            --lineage_path $db_name \
-            --cpu "${task.cpus}" \
-            --blast_single_core \
-            --mode genome \
-            --out ${assembly} \
-            >${assembly}_busco_log.txt
-        cp run_${assembly}/short_summary_${assembly}.txt short_summary_${assembly}.txt
-
-        for f in run_${assembly}/single_copy_busco_sequences/*faa; do 
-            [ -e "\$f" ] && cat run_${assembly}/single_copy_busco_sequences/*faa >${assembly}_buscos.faa || touch ${assembly}_buscos.faa
-            break
-        done
-        for f in run_${assembly}/single_copy_busco_sequences/*fna; do 
-            [ -e "\$f" ] && cat run_${assembly}/single_copy_busco_sequences/*fna >${assembly}_buscos.fna || touch ${assembly}_buscos.fna
-            break
-        done
-        """
-    } else {
-        """
-        run_BUSCO.py \
-            --in ${assembly} \
-            --lineage_path $db_name \
-            --cpu "${task.cpus}" \
-            --blast_single_core \
-            --mode genome \
-            --out ${assembly} \
-            >${assembly}_busco_log.txt
-        cp run_${assembly}/short_summary_${assembly}.txt short_summary_${assembly}.txt
-
-        for f in run_${assembly}/single_copy_busco_sequences/*faa; do 
-            [ -e "\$f" ] && cat run_${assembly}/single_copy_busco_sequences/*faa >${assembly}_buscos.faa || touch ${assembly}_buscos.faa
-            break
-        done
-        for f in run_${assembly}/single_copy_busco_sequences/*fna; do 
-            [ -e "\$f" ] && cat run_${assembly}/single_copy_busco_sequences/*fna >${assembly}_buscos.fna || touch ${assembly}_buscos.fna
-            break
-        done
-        """
-    }
+    # place db in extra folder to ensure BUSCO recognizes it as path (instead of downloading it)
+    mkdir dataset
+    mv ${db} dataset/
+
+    busco \
+        --mode genome \
+        --in ${bin} \
+        --lineage_dataset dataset/${db} \
+        --config \${config_file} \
+        --cpu "${task.cpus}" \
+        --out "BUSCO" > ${bin}_busco.log
```

Comment:
`--blast_single_core \` in busco3 was added to prevent blast problems when using multi cores, is that not a risk any more?

Reply:
yeah, that should be solved, it uses a blast version where it works

Reply:
Great!